helm/operator/templates/minio.min.io_tenants.yaml

```diff
@@ -875,6 +875,9 @@ spec:
                       type: string
                     type: array
                 type: object
+              certExpiryAlertThreshold:
+                format: int64
```

Comment:
Please use `int32` (it's in days, so `int64` is not needed). Marshalling 64-bit integers to Javascript is a pain, because it makes it a string. Using `int32` will keep it a number and keep things simple.

Reply:
I kept this as `int64` (1) in order to simplify the comparison between `certExpiryAlertThreshold` and `int64` var `expiresInDays`. Also, a separate helper function (2) would be needed to convert the possibly null (3)`*tenant.Spec.CertExpiryAlertThreshold` to `int64`, slowing down the program execution.
See https://github.com/minio/operator/blob/0b2bacb04bc0329eb1d16bfdfe7b8dd4ac1caa80/pkg/controller/custom.go#L111

In summary, using `int64` simplifies reading the code, reduces the amount of code needed, and does not affect the speed of execution as much.

Reply:
@ramondeklein Your thoughts on these reasons versus the marshalling complexity?

Reply:
@allanrogerr Good to have it as `int32`. I'll elaborate why I prefer this...

The problem with the default marshalling is that it sends `int64` protobuf via JSON as a `string`. Javascript handles each number as a 64-bit float (IEEE-754), where it uses 11 bits for the exponent. Because of that it cannot handle 64-bit integers with the proper precision and therefor marshalling reverts to using `string`. This requires both the Go and Javascript code to convert back/forth numbers. That's why I prefer using `int32` if it fits.

If it only saves a cast in Go from `int32` to `int64` (that will never loose precision), then I would prefer that instead. Converting from a string to a number always requires error-checking and dealing with the error. So if the semantics allow for `int32`, then I would use that instead.